src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -1659,11 +2115,793 @@ End Class
+            verifier.VerifyIL("Derived.M1()", source: source, sequencePoints: "Derived.M1", expectedIL: @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  // sequence point: M
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  ldvirtftn  ""string Derived.M()""
+  IL_0008:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000d:  ret
+}
+");
+            verifier.VerifyIL("Derived.M2()", source: source, sequencePoints: "Derived.M2", expectedIL: @"
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  // sequence point: base.M
+  IL_0000:  ldarg.0
+  IL_0001:  ldftn      ""object Base.M()""
+  IL_0007:  newobj     ""System.Func<object>..ctor(object, System.IntPtr)""
+  IL_000c:  ret
+}
+");
+            verifier.VerifyIL("Derived.M3()", source: source, sequencePoints: "Derived.M3", expectedIL: @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  // sequence point: new Func<string>(M)
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  ldvirtftn  ""string Derived.M()""
+  IL_0008:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000d:  ret
+}
+");
+            verifier.VerifyIL("Derived.M4()", source: source, sequencePoints: "Derived.M4", expectedIL: @"
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  // sequence point: new Func<object>(base.M)
+  IL_0000:  ldarg.0
+  IL_0001:  ldftn      ""object Base.M()""
+  IL_0007:  newobj     ""System.Func<object>..ctor(object, System.IntPtr)""
+  IL_000c:  ret
+}
+");
+            verifier.VerifyIL("Program.M1()", source: source, sequencePoints: "Program.M1", expectedIL: @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  // sequence point: M
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  ldvirtftn  ""string Derived.M()""
+  IL_0008:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000d:  ret
+}
+");
+            verifier.VerifyIL("Program.M2()", source: source, sequencePoints: "Program.M2", expectedIL: @"
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  // sequence point: base.M
+  IL_0000:  ldarg.0
+  IL_0001:  ldftn      ""string Derived.M()""
+  IL_0007:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000c:  ret
+}
+");
+            verifier.VerifyIL("Program.M3()", source: source, sequencePoints: "Program.M3", expectedIL: @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  // sequence point: new Func<string>(M)
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  ldvirtftn  ""string Derived.M()""
+  IL_0008:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000d:  ret
+}
+");
+            verifier.VerifyIL("Program.M4()", source: source, sequencePoints: "Program.M4", expectedIL: @"
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  // sequence point: new Func<string>(base.M)
+  IL_0000:  ldarg.0
+  IL_0001:  ldftn      ""string Derived.M()""
+  IL_0007:  newobj     ""System.Func<string>..ctor(object, System.IntPtr)""
+  IL_000c:  ret
+}
+");
         }
```

Comment:
>} [](start = 8, length = 1)

It feels like we should test how overriding related APIs behave in constructed types.